website/blog/2023-11-15-run-webassembly-wasm-workloads-windows-and-macos.md

```diff
@@ -0,0 +1,339 @@
+
+The binary format serves as a compilation target and it allows to use a wide ranger of programming languages such as C, C++, and Rust. While it was a browser/client technology, now it is evolving beyond the web, for example being adapted for use as a back-end or edge technology. (This is for example what happened to Java that was first designed for the client side before landing to the server side)
+
+The Wasm binary format was designed to be secure. Wasm modules are isolated from the rest of the system, and they cannot access any system resources without explicit permission. This makes Wasm modules very safe to run, even in untrusted environments. But in another hand, for developing backend applications, this restriction is limiting the usage of WASM.
+
+## The extension of WebAssembly
+
+Then WebAssembly System Interface (WASI) was born as an essential complement to WebAssembly.
+
+It is a system interface that extends WebAssembly's capabilities beyond the browser, making it suitable for a wide range of environments, including servers, edge devices, and more.
+
+While with WASM you had a limited access to the host resources, WASI provides a standard set of system calls, enabling WebAssembly modules to interact with the host operating system in a secure and consistent manner: it includes filesystem access, sockets, and other low-level resources.
+
+## Running WebAssembly outside the web
+
+Wasm has shipped in the major browser engines so the usage of wasm is possible without any 3rd party addition in the browser land. But when it comes to the edge/system usage, you need to find a virtual machine to run these workloads supporting WASI extension. And there is not only one application to run them, there are several wasm runtime such as WasmEdge, Wasmtime, Wasmer, and so on. All runtimes support different CPU architectures.
+
+Due to the maturity of WASI that is still young you might have API provided in each of these runtime that has not yet reached the standard so users need to be careful to write portable applications that do not depend on a given runtime.
+
+While you can run wasm/WASI workloads on your computer,  how to you package this binary format, share it and distribute it. A convenient way to distribute and run these workloads is to use OCI images as it provides all the basics: package, storage and distribution of the binaries. Then comes the execution part.
+
+## Using Podman engine with wasm
+
+When using containers with Podman on macOS or Windows, you have a virtual machine called a "Podman machine" that is executing a Linux environment. We need to add a runtime inside this Linux box. Podman is using as OCI runtime the crun project so this component needs to be able to run or delegate the execution to WASM runtimes. Good enough for us, crun supports the WASM execution.
```

Comment:
'We need to add a runtime inside this Linux box. Podman is using as OCI runtime the crun project so this component needs to be able to run or delegate the execution to WASM runtimes. Good enough for us, crun supports the WASM execution.'
Suggested rephrasing:
'We need to add support for WASM inside this Linux environment. Podman is using the run project as its OCI runtime, so crun needs to be able to run or delegate execution to WASM runtimes. Lucky for us, crun supports WASM execution.'